src/components/MessageList/TypingIndicator.js

```diff
@@ -24,74 +22,72 @@ const Container = styled.View`
   ${({ theme }) => theme.typingIndicator.container.css};
 `;
 
-class TypingIndicator extends React.PureComponent {
-  static themePath = 'typingIndicator';
-  static propTypes = {
-    typing: PropTypes.object,
-    client: PropTypes.instanceOf(StreamChat),
-    /**
-     * Defaults to and accepts same props as: [Avatar](https://getstream.github.io/stream-chat-react-native/#avatar)
-     */
-    Avatar: PropTypes.oneOfType([PropTypes.node, PropTypes.elementType]),
-  };
-  static defaultProps = {
-    Avatar,
-  };
+const constructTypingString = (typingUsers, client, t) => {
+  const typingKeys = Object.keys(typingUsers);
+  const nonSelfUsers = [];
+  typingKeys.forEach((item, i) => {
+    if (client.user.id === typingUsers[typingKeys[i]].user.id) {
+      return;
+    }
+    nonSelfUsers.push(
+      typingUsers[typingKeys[i]].user.name ||
+        typingUsers[typingKeys[i]].user.id,
+    );
+  });
 
-  constructTypingString = (dict) => {
-    const { client, t } = this.props;
-    const arr2 = Object.keys(dict);
-    const arr3 = [];
-    arr2.forEach((item, i) => {
-      if (client.user.id === dict[arr2[i]].user.id) {
-        return;
-      }
-      arr3.push(dict[arr2[i]].user.name || dict[arr2[i]].user.id);
+  let outStr = '';
+  if (nonSelfUsers.length === 1) {
+    outStr = t('{{ user }} is typing...', { user: nonSelfUsers[0] });
+  } else if (nonSelfUsers.length === 2) {
+    //joins all with "and" but =no commas
+    //example: "bob and sam"
+    outStr = t('{{ firstUser }} and {{ secondUser }} are typing...', {
+      firstUser: nonSelfUsers[0],
+      secondUser: nonSelfUsers[1],
+    });
```

Comment:
I've noticed this string replacement style in a few places. is that necessary for how `t` works with translating or could we use these "`" instead. wondering for consistency of string replacements throughout the code

Reply:
AFAIK template literals don't work for i18n because of the interpolation. If it's a non interpolated string I think it will work, i.e. t(`Hello`) but otherwise it string matches the {{ syntax in the original and translated string. Is that what you are reffering to?

Reply:
gotcha, yeah that's what I was referring to